src/theme/Mermaid.js

```diff
@@ -43,11 +44,19 @@ const Mermaid = ({ chart }) => {
   const [svg, setSvg] = useState(undefined)
   const [id] = useState(`mermaid-${Math.random().toString(36).substr(2, -1)}`)
   const toggle = () => setZoomed(!zoomed)
+  const { colorMode } = useColorMode()
 
   useEffect(() => {
-    mermaid.render(id, chart, (svg) => {
-      setSvg(svg)
-    })
+    // https://mermaid.js.org/config/theming.html#diagram-specific-themes
+    mermaid.render(
+      id,
+      `%%{init: {'theme':'${
+        colorMode === "light" ? "neutral" : "dark"
+      }'}}%%\n${chart}`,
+      (svg) => {
+        setSvg(svg)
+      },
+    )
   }, [])
```

Comment:
It should be enough to add the color mode here to not require reloads.
```suggestion
  }, [colorMode])
```

Reply:
Ok, so let's merge as is and we'll figure out the rest later. It is a huge improvement anyway, as @vinckr said. Thanks again! :tada: